### DISCUSSION TITLE
Подбор товара через Селект-дерево в заказ (цикл)

### DISCUSSION THREAD

**Discussion by DmitriyGulevichS** [totumonline/totum-ru-issues-and-discussions#518] [Как это сделать?] [answered]
Добрый день! Вопрос следующий, как реализовать подбор товаров через дерево в заказ (таблица циклов). 

**Иерархия такая:** 

Категория > Подкатегория > Товар. 

**Пример:** 
 
Тумбы > Тумбы с ящиками > Тумба нижняя 960х560х500 
Шкафы > Шкафы навесные > Шкаф навесной 920x400x100 

При клике на кнопку в заказе "Подбор" должна открываться таблица с деревом, пользователь выбирает необходимые товары количеством или чек-боксом и нажимает кнопку "Добавить в заказ". После модалка закрывается и  товары добавляются в заказ. 

Принцип как у вас в в готовом примере с заказами и складом, но там подбор через временную таблицу реализован. Это я смог сделать, а вот с деревом не смог разобраться. 

PS. минусы решения которое в примерах, нет префильтра во временной (реализовал через фильтр по полю). когда товаров много, нужно их как то делить. Возможно какими то табами в хедере, но в идеале иерархия и выбор товара через дерево) 

Заранее спасибо!

**Comment by erpic-ru** [comment]
А что вам мешает добавить дерево из простой во временную таблицу?
Посмотрите, [через автозаполнение](https://ru.docs.totum.online/autofill-tables) все работает
<img width="1603" alt="Снимок экрана 2024-02-27 в 17 37 20" src="https://github.com/totumonline/totum-ru-issues-and-discussions/assets/91674114/fee5139f-40c3-4875-bc17-7520788fdc04">

**Reply by DmitriyGulevichS** [reply]
Вот фактически мне нужно то, что у вас на скриншоте. 
Я понимаю как работает автозаполнение и могу его реализовать) Но именно с деревом не понимаю как работать, если есть подробная инструкция как это сделать, я бы попробовал. На скрине оч круто реализовано! Хочу)))

PS
Я смотрел видео в обучающем курсе, но так и не понял как.

**Reply by DmitriyGulevichS** [reply]
![image](https://github.com/totumonline/totum-ru-issues-and-discussions/assets/100049818/44716d7e-f615-43d2-8754-4f565c2073a8)

**Reply by erpic-ru** [reply, marked as answer]
Вот выгрузка с простеньким деревом во временной. Загрузите куда-то сбоку, возможно разберетесь
- Структура номенклатуры
- Номенклатура
- Временная с выбором позиций

[tree.json.gz.ttm.zip](https://github.com/totumonline/totum-ru-issues-and-discussions/files/14443008/tree.json.gz.ttm.zip)

**Comment by DmitriyGulevichS** [comment]
![image](https://github.com/totumonline/totum-ru-issues-and-discussions/assets/100049818/1dc67948-22ef-43df-8a3e-961ef21afd13)

 Спасибо тебе огромное дружище! Разобрался!
